components/eventing-controller/pkg/backend/jetstreamv2/jetstream_integration_test.go

```diff
@@ -910,6 +910,142 @@ func TestJSSubscriptionUsingCESDK(t *testing.T) {
 	require.NoError(t, jsBackend.DeleteSubscription(sub))
 }
 
+// TestJetStreamSubAfterSync_DeleteOldFilterConsumerForFilterChangeWhileNatsDown tests the SyncSubscription method
+// when subscription CR filters change while NATS JetStream is down.
+func TestJetStreamSubAfterSync_DeleteOldFilterConsumerForFilterChangeWhileNatsDown(t *testing.T) {
```

Comment:
this function has multiple `given` / `when` /`then` parts in it. could it be changed in a way that makes it easier to understand?